polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -1233,3 +1271,16 @@ fn pvf_exec_timeout(executor_params: &ExecutorParams, kind: PvfExecKind) -> Dura
 		PvfExecKind::Approval => DEFAULT_APPROVAL_EXECUTION_TIMEOUT,
 	}
 }
+
+fn pvf_exec_deadline(
+	exec_timeout: Duration,
+	validation_request_ttl: Option<Duration>,
+) -> Option<Instant> {
+	validation_request_ttl.and_then(|ttl| {
+		if ttl <= exec_timeout {
+			None // Job dropping should be turned off
+		} else {
+			Some(Instant::now() + ttl - exec_timeout)
```

Comment:
I don't think we should substract this here, there is a good chance your backing could still make it if execution is fast, but this assumes all executions are `exec_timeout` long and it shouldn't since that's the worse case scenario.

Reply:
You're right, with fast execution, we can easily miss a candidate. I share the same concerns. However, we might waste `exec_timeout` trying to execute it on time. Since it's part of the back-pressure mechanism where we delay and skip backing, we can use this extra time to handle disputes and approvals, which will save finality. So that's a very reasonable question. 
@sandreim WDYT?

Reply:
Better err on the side of a little wasted execution, otherwise this will cause issues at some point.

Reply:
> You're right, with fast execution, we can easily miss a candidate. I share the same concerns. However, we might waste exec_timeout trying to execute it on time. Since it's part of the back-pressure mechanism where we delay and skip backing, we can use this extra time to handle disputes and approvals, which will save finality. So that's a very reasonable question.

I'm confused the prioritisation part should already handle this because we prefer disputes/approvals over backing jobs, from what I understood this PR is for handling the case where because we pick backing jobs so rarely when we are overloaded we could end up in a situation where the backing job we picked is already past ttl so you are in kind of a negative feedback loop where you keep checking backing candidates that have no chance of making it in the relay-chain.

The good part is that this helps with the congestion, since it reduces the number of candidates put on chain and give time for approvals to catch up.

Reply:
I share the same concerns of this affecting block times when there isn't enough pressure.

> Better err on the side of a little wasted execution, otherwise this will cause issues at some point.

Yeah, we can just assume fastest execution to preserve liveness. But it could also make it smarter if we wanted. We could computed a dynamic value for the estimated execution time based on the minimum execution time observed in the past for each parachain.